be-postgres.c

```diff
@@ -182,6 +182,12 @@ int be_pg_getuser(void *handle, const char *username, const char *password, char
 
 	if (PQresultStatus(res) != PGRES_TUPLES_OK) {
 		_log(LOG_DEBUG, "%s\n", PQresultErrorMessage(res));
+		if(PQstatus(conf->conn) == CONNECTION_BAD){
```

Comment:
What if this was executed prior to running a query - will that still work?

Reply:
I am not sure if i understand your comment correctly. How should that be possible at that position ? I understand that all postgres queries are currently performed synchronously. That means that the query has been finished (not sucessfully) when this piece of code comes into play.

On line 174 there is a check if the database connection has been initialized 

```
if (!conf || !conf->userquery || !username || !*username)
		return BACKEND_DEFER;
```

Reply:
I've looked at the source, and you are of course correct.